### PR TITLE
Adds `svg-has-accessible-name` rule

### PR DESCRIPTION
**Pull request by lindseywild**
Relates to https://github.com/github/accessibility/issues/3972.

### Description

Adds a lint rule to ensure SVGs have an accessible name by checking for a title element as the first descendant of the `<svg>` element or an `aria-label` or `aria-labelledby`. It also checks if the element is presentational, in which an error won't be thrown.

### CODE REVIEW

File: lib/rules/a11y-svg-has-accessible-name.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  create(context) {
+    return {
+      JSXOpeningElement: node => {
+        const elementType = getElementType(context, node)
+
+        // Check if there is a nested title element that is the first child of the `<svg>`
+        const hasNestedTitleAsFirstChild =
+          node.parent.children.length &&
+          node.parent.children[0].type === 'JSXElement' &&
+          node.parent.children[0].openingElement.name.name === 'title'
```
Comment (keithamus) [inline lib/rules/a11y-svg-has-accessible-name.js:22]: May want to guard these which will prevent errors on property access:

```suggestion
        const hasNestedTitleAsFirstChild =
          node.parent.children?.[0]?.type === 'JSXElement' &&
          node.parent.children?.[0]?.openingElement?.name?.name === 'title'
```

File: lib/rules/a11y-svg-has-accessible-name.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      description: 'SVGs must have an accessible name',
+      url: require('../url')(module),
+    },
+    schema: [],
+  },
+
+  create(context) {
+    return {
+      JSXOpeningElement: node => {
+        const elementType = getElementType(context, node)
```
Comment (keithamus) [inline lib/rules/a11y-svg-has-accessible-name.js:16]: An early return here will speed up execution which can get bogged down with many small cuts:

```suggestion
        const elementType = getElementType(context, node)
        if (elementType !== 'svg') return
```

**Review by keithamus — APPROVED** [review]
Very nice :+1:

File: tests/a11y-svg-has-accessible-name.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+const rule = require('../lib/rules/a11y-svg-has-accessible-name')
```
Comment (khiga8) [inline tests/a11y-svg-has-accessible-name.js:1]: nice tests!! thank you!!

File: docs/rules/a11y-svg-has-accessible-name.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+💼 This rule is enabled in the ⚛️ `react` config.
+
+<!-- end auto-generated rule header -->
+
+## Rule Details
+
+An `<svg>` must have an accessible name. Set `aria-label` or `aria-labelledby`, or nest a `<title>` element as the first child of the `<svg>` element.
+
+However, if the `<svg>` is purely decorative, hide it with `aria-hidden="true"` or `role="presentation"`.
```
Comment (khiga8) [inline docs/rules/a11y-svg-has-accessible-name.md:11]: Ahhh I didn't consider `role="presentation"`! TIL!

**Review by khiga8 — APPROVED** [review]
This looks great! Let's ship ittttt 🚀

**Pull request merged**